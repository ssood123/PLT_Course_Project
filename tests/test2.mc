function void printOneMore()
{
  printStr("Last");
}


function void printMore()
{
  print(108);
  printf(99.9-99.81);
  printStr("New");
  printOneMore();
}


function void main()
{
  int a;
  float b;
  string c;
  a = 110 + 120;
  b = 110.0253 * 56.7;
  c = "thisIsATest";
  print(a);
  printf(b);
  printStr(c);
  printMore();
}