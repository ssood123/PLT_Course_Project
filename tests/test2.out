; ModuleID = 'integraph'
source_filename = "integraph"

@fmt = private unnamed_addr constant [4 x i8] c"%d\0A\00", align 1
@fmt.1 = private unnamed_addr constant [4 x i8] c"%s\0A\00", align 1
@fmt.2 = private unnamed_addr constant [4 x i8] c"%g\0A\00", align 1
@tmp = private unnamed_addr constant [12 x i8] c"thisIsATest\00", align 1
@fmt.3 = private unnamed_addr constant [4 x i8] c"%d\0A\00", align 1
@fmt.4 = private unnamed_addr constant [4 x i8] c"%s\0A\00", align 1
@fmt.5 = private unnamed_addr constant [4 x i8] c"%g\0A\00", align 1
@tmp.6 = private unnamed_addr constant [4 x i8] c"New\00", align 1
@fmt.7 = private unnamed_addr constant [4 x i8] c"%d\0A\00", align 1
@fmt.8 = private unnamed_addr constant [4 x i8] c"%s\0A\00", align 1
@fmt.9 = private unnamed_addr constant [4 x i8] c"%g\0A\00", align 1
@tmp.10 = private unnamed_addr constant [5 x i8] c"Last\00", align 1

declare i32 @printf(i8*, ...)

declare i32 @printbig(i32)

define void @main() {
entry:
  %a = alloca i32, align 4
  %b = alloca double, align 8
  %c = alloca i8*, align 8
  store i32 230, i32* %a, align 4
  store double 0x40B85E6F3C0C1FC9, double* %b, align 8
  store i8* getelementptr inbounds ([12 x i8], [12 x i8]* @tmp, i32 0, i32 0), i8** %c, align 8
  %a1 = load i32, i32* %a, align 4
  %printf = call i32 (i8*, ...) @printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @fmt, i32 0, i32 0), i32 %a1)
  %b2 = load double, double* %b, align 8
  %printf3 = call i32 (i8*, ...) @printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @fmt.2, i32 0, i32 0), double %b2)
  %c4 = load i8*, i8** %c, align 8
  %printf5 = call i32 (i8*, ...) @printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @fmt.1, i32 0, i32 0), i8* %c4)
  call void @printMore()
  ret void
}

define void @printMore() {
entry:
  %printf = call i32 (i8*, ...) @printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @fmt.3, i32 0, i32 0), i32 108)
  %printf1 = call i32 (i8*, ...) @printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @fmt.5, i32 0, i32 0), double 0x3FB70A3D70A3D800)
  %printf2 = call i32 (i8*, ...) @printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @fmt.4, i32 0, i32 0), i8* getelementptr inbounds ([4 x i8], [4 x i8]* @tmp.6, i32 0, i32 0))
  call void @printOneMore()
  ret void
}

define void @printOneMore() {
entry:
  %printf = call i32 (i8*, ...) @printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @fmt.8, i32 0, i32 0), i8* getelementptr inbounds ([5 x i8], [5 x i8]* @tmp.10, i32 0, i32 0))
  ret void
}
